File: services/gateway/app/engine/chunking.py
"""File chunking for analysis (mirrors services/engine chunking)."""
from __future__ import annotations


def split_files(file_paths: list[str], *, ignore_lockfiles: bool = True) -> list[str]:
    out: list[str] = []
    for path in file_paths:
        if ignore_lockfiles and (
            "package-lock" in path or "pnpm-lock" in path or path.endswith("yarn.lock")
        ):
            continue
        out.append(path)
    return out or file_paths[:1] or ["unknown"]

File: services/engine/src/chunking/chunking.cpp
#include <string>
#include <vector>

namespace prism::chunking {

std::vector<std::string> split_files(const std::vector<std::string>& paths, bool ignore_lockfiles) {
  std::vector<std::string> out;
  for (const auto& p : paths) {
    if (ignore_lockfiles && (p.find("package-lock") != std::string::npos ||
                             p.find("pnpm-lock") != std::string::npos)) {
      continue;
    }
    out.push_back(p);
  }
  return out;
}

}  // namespace prism::chunking
